core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -655,7 +656,16 @@ public void setDefaultTTL(int defaultTTL) {
                         .doOnNext(i -> insertedDataPointEvents.onNext(metric)))
                 .doOnNext(meter::mark);
 
-        Observable<Integer> indexUpdates = dataAccess.updateMetricsIndex(metrics)
+        Observable<Metric<T>> updateIdxMetrics = metrics.filter(metric -> {
```

Comment:
The problem with enhancing this with further commits is that the first thing to do is revert this commit (this does not provide any starting point for the functionality @jsanda is thinking about).